File: ruby/merge_sorted_array/spec/merge_sorted_array_spec.rb
require './merge_sorted_array.rb'

describe 'merge sorted array' do
  it 'merges two arrays in ascending order' do
    # arrange
    nums1 = [1,2,3,0,0,0]
    nums2 = [2,5,6]

    # act
    result = merge(nums1, nums2)

    # assert
    expect(result).to eq([1,2,2,3,5,6])
  end
end

File: haskell/merge_sorted_array/MergeSortedArraySpec.hs
module MergeSortedArraySpec where
import MergeSortedArray
import Test.Hspec

main = hspec $ do
  describe "merge" $ do
    it "merges two arrays" $ do
      merge [1,2,3,0,0,0] [2,5,6]
        `shouldBe` [1,2,2,3,5,6]
